File: backend/src/Workora.Application/Features/Training/DTOs/TrainingDto.cs
using Workora.Domain.Enums;

namespace Workora.Application.Features.Training.DTOs;

/// <summary>
/// DTO representing a training program course.
/// </summary>
public record TrainingProgramDto(
    int Id,
    Guid Uuid,
    int CompanyId,
    string Title,
    string Description,
    string TrainerName,
    DateOnly StartDate,
    DateOnly EndDate,
    int Capacity,
    int EnrolledCount,
    bool IsActive,
    DateTimeOffset CreatedAt);

/// <summary>
/// DTO representing a training enrollment.
/// </summary>
public record TrainingEnrollmentDto(
    int Id,
    int TrainingProgramId,
    string? ProgramTitle,
    int EmployeeId,
    string? EmployeeName,
    TrainingStatus Status,
    DateTimeOffset? CompletedAt,
    DateTimeOffset CreatedAt);

/// <summary>
/// Request payload for creating a training program.
/// </summary>
public record CreateTrainingProgramRequestDto(
    int CompanyId,
    string Title,
    string Description,
    string TrainerName,
    DateOnly StartDate,
    DateOnly EndDate,
    int Capacity);

/// <summary>
/// Request payload for enrolling an employee.
/// </summary>
public record EnrollTrainingRequestDto(
    int TrainingProgramId,
    int EmployeeId);

File: frontend/workora-ui/src/app/data/dtos/training.dto.ts
export interface TrainingProgramDto {
  id: number;
  uuid: string;
  companyId: number;
  title: string;
  description: string;
  trainerName: string;
  startDate: string;
  endDate: string;
  capacity: number;
  enrolledCount: number;
  isActive: boolean;
  createdAt: string;
}

export interface TrainingEnrollmentDto {
  id: number;
  trainingProgramId: number;
  programTitle?: string | null;
  employeeId: number;
  employeeName?: string | null;
  status: string;
  completedAt?: string | null;
  createdAt: string;
}

export interface CreateTrainingProgramRequestDto {
  companyId: number;
  title: string;
  description: string;
  trainerName: string;
  startDate: string;
  endDate: string;
  capacity: number;
}

export interface EnrollTrainingRequestDto {
  trainingProgramId: number;
  employeeId: number;
}
